src/Table.cpp

```diff
@@ -0,0 +1,102 @@
+//
+// Created by derekjtong on 4/17/2023.
+//
+#include "sudoManager/Table.hpp"
+
+#include <stdexcept>
+Table::Table(int numberOfSeats) {
+  for (int i = 0; i < numberOfSeats; i++) {
+    m_seats.push_back(TableSeat(i));
+  };
+}
+int Table::getCapacity() const { return m_seats.size(); }
+
+bool Table::isOccupied() const {
+  for (auto& seat : m_seats)
+    if (seat.isOccupied()) return true;
+  return false;
+}
+
+bool Table::isServed() const {
+  for (auto& seat : m_seats) {
+    if (seat.isOccupied() && !seat.isServed()) {
+      return false;
+    }
+  }
+  return true;
+}
+
+int Table::getUnservedSeats() const {
+  int unservedSeats = 0;
+  for (auto& seat : m_seats) {
+    if (seat.isOccupied() && !seat.isServed()) {
+      unservedSeats++;
+    }
+  }
+  return unservedSeats;
+}
+
+void Table::occupyAll() {
+  if (isOccupied()) {
+    throw std::logic_error("Error: Table is already occupied");
+  }
+  for (auto& seat : m_seats) {
+    seat.setOccupied(true);
+  }
```

Comment:
```suggestion
    m_seats.occupy(m_seats.size());
```
:wink:

Reply:
`occupy(m_seats.size());`
Sorry for my mistake.

unless there is parallelism involved I would choose ranged-for over `for_each`